### ISSUE TITLE
Hystrix With ExecutorService in Spring Boot

### ISSUE THREAD

**Issue by amarflybot** [spring-cloud/spring-cloud-netflix#3618]
I have a new service in Spring Boot. To Serve one request the service has to go to collect data from multiple sources with Executor Service in place. How do I use Hystrix in between this service and downstream service.

Lets say I have a SERVICE A that needs to call SERVICE B twice and the result from SERVICE B is needed to call SERVICE C 200 times, before it can response back. I have a Executor Service to Call services in parallel.

I have tried @HystrixCommand with SEMAPHORE Configuration for all service calls in between SERVICE A, SERVICE B and SERVICE C.

By time I the thread gets scheduled for SERVICE C, Hystrix kicks off and some of the 200 calls to SERVICE C from SERVICE A gets tripped.

Response time expected by SERVICE A : 1 sec Response time of SERVICE B: 300 ms Response time of SERVICE C: 40-100 ms

AWS Beanstalk Machine Used has 8 core CPU.

I have also used a redis Cache to cache the responses of SERVICE C. But SERVICE B call cannot be cached, it has dynamic result.

Executor Service Config:
`@Bean
    public Executor asyncGCPExecutor() {
        final ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
        executor.setCorePoolSize(30);
        executor.setMaxPoolSize(50);
        executor.setThreadNamePrefix("async-");
        executor.setKeepAliveSeconds(10);
        executor.setWaitForTasksToCompleteOnShutdown(true);
        executor.initialize();
        return executor;
    }
`

Hystrix Config:
`hystrix.command.default.fallback.isolation.semaphore.maxConcurrentRequests=100
hystrix.command.default.execution.isolation.strategy=SEMAPHORE
hystrix.command.default.execution.isolation.thread.timeoutInMilliseconds=600
`

Tries a lot with different number of thread configs and timeout but it did not help. Please guide me with the correct hystrix config.

Please let me know if the situation is unclear. Link for the Image: https://imgur.com/a/Pq2VP69

StackOverFlow URL: https://stackoverflow.com/questions/57331135/hystrix-with-executorservice-in-spring-boot

**Comment by amarflybot** [comment]
This is mostly a question ticket.

**Comment by spencergibb** [comment]
Please don't cross post.

Closing in favor of stack overflow.
